jvb/src/main/kotlin/org/jitsi/videobridge/Endpoint.kt

```diff
@@ -1239,4 +1359,379 @@ class Endpoint @JvmOverloads constructor(
+        }
+
+        /**
+         * Send all current mappings to the endpoint.
+         * Can be used to resynchronize after message transport reconnects.
+         */
+        fun sendAllMappings() {
+            if (mediaType == MediaType.VIDEO) {
+                val remappings = mutableListOf<VideoSourceMapping>()
+
+                synchronized(sendSources) {
+                    sendSources.forEach { _, sendSource ->
+                        val props = sendSource.props
+                        val mapping = VideoSourceMapping(
+                            props.name, props.owner, sendSource.send1.ssrc, sendSource.send2.ssrc, props.videoType
+                        )
+                        remappings.add(mapping)
+                    }
+                }
+
+                if (!remappings.isEmpty())
+                    sendMessage(VideoSourcesMap(remappings))
+            } else if (mediaType == MediaType.AUDIO) {
+                val remappings = mutableListOf<AudioSourceMapping>()
+
+                synchronized(sendSources) {
+                    sendSources.forEach { _, sendSource ->
+                        val props = sendSource.props
+                        val mapping = AudioSourceMapping(props.name, props.owner, sendSource.send1.ssrc)
+                        remappings.add(mapping)
+                    }
+                }
+
+                if (!remappings.isEmpty())
+                    sendMessage(AudioSourcesMap(remappings))
+            }
+        }
+
+        /**
+         * Rewrite RTP fields for a relayed packet.
+         * Activates a send SSRC if necessary.
+         * @param packet the packet about to be sent.
+         * @param start whether this packet can be the first packet sent on a new SSRC mapping.
+         * @return whether to send this packet.
+         */
+        fun rewriteRtp(packet: RtpPacket, start: Boolean = true): Boolean {
+
+            var send: Boolean = false
+
+            logger.debug { "$mediaType packet: ${debugInfo(packet)}" }
+
+            synchronized(sendSources) {
+                var rs = receivedSsrcs.get(packet.ssrc)
+                if (rs == null) {
+                    val props = findSourceProps(packet.ssrc, mediaType)
+                    if (props == null) {
+                        return false
+                    }
+                    rs = ReceiveSsrc(props)
+                    receivedSsrcs.put(packet.ssrc, rs)
+                    logger.debug { "added $mediaType receive SSRC: ${packet.ssrc}" }
+                }
+
+                val ss = getSendSource(rs.props.ssrc1, rs.props, mediaType == MediaType.AUDIO, null)
+                if (ss != null) {
+                    send = ss.rewriteRtp(packet, start, rs)
+                    logger.debug { this.toString() }
+                    logger.debug {
+                        if (send) {
+                            "output packet: ${debugInfo(packet)} source=${rs.props.name} start=$start send=$send"
```

Comment:
Log messages should generally be capitalized

Reply:
fixed.